### PR TITLE
General security updates for application dependencies

### PR DESCRIPTION
**Pull request by jpl-jengelke**
- Fix for Python 3.12 incompatibilities
- Upgrade following modules:
> +astroalign~=2.6.0
+astropy~=6.1
+colour_demosaicing~=0.2.6
+matplotlib~=3.9.2
+numpy~=1.26.4
+pandas~=2.2.3
+panel~=1.5.2
+photutils~=2.0.0
+pyvo~=1.5.3
+rebound~=4.4.3
+scipy~=1.14.1

- Tested with `HAT-P-32 b`
- Not yet tested/vetted on Colab

### CODE REVIEW

**Comment by tamimfatahi** [comment]
Let's factor in issue #1254 as well. I'll add it onto the branch.

File: requirements.txt [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
 imreg_dft~=2.0.0
 LDTk~=1.8.4
 lmfit~=1.3.2
-matplotlib~=3.7.5
-numpy~=1.24.4
-pandas~=1.5.3
-panel~=1.4.5
-photutils~=1.13.0
+matplotlib~=3.9.2
+numpy~=1.26.4
```
Comment (tamimfatahi) [inline requirements.txt:13]: Something worth keeping an eye out on the future is updating to bumpy 2.0 once our other dependents do so.

**Review by tamimfatahi — APPROVED** [review]
Tested on a couple of datasets and passed using Python 3.10.

File: requirements.txt (hunk 1 above)
Comment (jpl-jengelke) [reply requirements.txt:13]: Most of them are Numpy 2.x compatible now. But we couldn't update Numpy due to an inconsistency with LDTk. So the Numpy version was rolled back to the most recent that has not caused an issue. Numpy 1.26.4 is the last in the 1.x line and released in February 2024. Until LDTk is updated (or we can perhaps update Numba and fix another package hdbscan==0.8.39) any Numpy update will be inconsistent. It will also require deleting some cached HDB files to reset them. 
``` 
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject```

Bottom line is it's the best we can do for now, I think.

**Comment by ivenzor** [comment]
Tested a dataset on Windows with the new requirements, everything good.
